### DISCUSSION TITLE
Possibility to imprint a written message on blockchain

### DISCUSSION THREAD

**Discussion by Oscillator098** [Chia-Network/chia-blockchain#8395] [Q&A] [answered]
Hi,
I'm wondering is it possible to imprint/engrave a written message in chia blockchain to permanently stay there?
Usecase is that you can verify a statement that is written sometime in the past.

Even more so, this can be probably done trough NFT and smart contracts, but I don't know does it exist; can 2 parties (lets say 2 Chia wallets) sign on some form of written contract that is confirmed on blockchain?

**Comment by loppefaaret** [comment, marked as answer]
im sure you can create a tool that saves a string of text into a transaction
there is also `chia keys sign` and `chia keys verify` for signing messages if that is any help ?
